clients/command/src/main/java/org/eclipse/hono/client/command/ProtocolAdapterCommandConsumerFactoryImpl.java

```diff
@@ -213,26 +220,22 @@ private Future<CommandConsumer> doCreateCommandConsumer(
         final Instant lifespanStart = Instant.now();
 
         return commandRouterClient
-                .registerCommandConsumer(tenantId, deviceId, adapterInstanceId, sanitizedLifespan, context)
+                .registerCommandConsumer(tenantId, deviceId, sendEvent, adapterInstanceId, sanitizedLifespan, context)
                 .onFailure(thr -> {
-                    LOG.info("error registering consumer with the command router service [tenant: {}, device: {}]",
-                            tenantId, deviceId, thr);
+                    LOG.info(
+                            "error registering consumer with the command router service [tenant: {}, device: {}, sendEvent: {}]",
+                            tenantId, deviceId, sendEvent, thr);
                     // handler association failed - unregister the handler
                     commandHandlers.removeCommandHandler(tenantId, deviceId);
                 })
-                .map(v -> {
-                    return new CommandConsumer() {
-                        @Override
-                        public Future<Void> close(final SpanContext spanContext) {
-                            return removeCommandConsumer(commandHandlerWrapper, sanitizedLifespan,
-                                    lifespanStart, spanContext);
-                        }
-                    };
-                });
+                .map(v -> (ProtocolAdapterCommandConsumer) (sendEvent1, spanContext) -> removeCommandConsumer(
+                        commandHandlerWrapper, sendEvent1, sanitizedLifespan,
+                        lifespanStart, spanContext));
```

Comment:
I think keeping the anonymous class instead of using a lambda makes this more readable.

Reply:
agree, improved.